envoy/config/subscription.h

```diff
@@ -158,6 +158,17 @@ class UntypedConfigUpdateCallbacks {
   virtual void onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt::Any>& resources,
                               const std::string& version_info) PURE;
 
+  /**
+   * Called when a non-delta gRPC configuration update is received.
```

Comment:
Why do we have two `onConfigUpdate` for non-delta with different signatures now?

Reply:
I added it so we don't decode resources twice: before this change we decoded a resource to handle ttl, then used its non-decoded representation in onConfigUpdate invocation, where it was decoded the second time. That lead to ~30% slower eds_speed_test results. With the method I added the results are on par.

Reply:
Makes sense, but could we get rid of the above method on line 158 that is now redundant?

Reply:
We discussed this offline; the method on line 158 can be removed once the legacy muxes are gone. I'd say just leave a TODO and/or bug and we can wrap this.

LGTM, @alyssawilk any other concerns before merging?

Reply:
I think we lost the fix to test/per_file_coverage.sh at some point and I'd like to see that fixed before this goes in.

Reply:
> I think we lost the fix to test/per_file_coverage.sh at some point and I'd like to see that fixed before this goes in.

@alyssawilk: not sure what you mean by that -- this PR no longer makes any changes to `test/per_file_coverage.sh`.

Reply:
Right.  In your prior PR you added an exemption to test/per_file_coverage.sh, commiting code which did not abide by the Envoy test coverage standards.  I asked that you fix that in the follow up, and I would prefer we not continue merging changes here without the code meeting the Envoy testing bar.  You can show that it does by removing the exemption you previously added to test/per_file_coverage.sh